layer/contracts/runs.py

```diff
@@ -198,53 +176,8 @@ def _pack(self) -> None:
         with open(self.pickle_path, mode="wb") as file:
             cloudpickle.dump(self.func, file, protocol=pickle.DEFAULT_PROTOCOL)
 
-        # Add requirements to tarball
-        if self.pip_requirements_file:
-            shutil.copy(self.pip_requirements_file, self.environment_path)
-        elif self.pip_packages:
-            with open(self.environment_path, "w") as reqs_file:
-                reqs_file.writelines(
-                    list(map(lambda package: f"{package}\n", self.pip_packages))
-                )
```

Comment:
you seem to be changing this part as part of this refactor -- did you test this thoroughly?

Reply:
not sure what is thoroughly, but tested with several remote execution examples that I think should cover this.